### PR TITLE
Keep 3 days of nightlies

### LINKED ISSUE #472: Keep 3 days worth of nightlies
We should keep 3 days worth of nightlies. Each nightly will have a new tag (like `nightly-${commitish}`) in addition to the `nightly` tag.

Unfortunately we can't download off of the `nightly` tag unless it is the main tag used for the release, but we can't have it be the main tag either, since that would just override the previous nightly release.

So we need to:

1. Find the commit that the  `nightly` tag is pointing to
2. Lookup the release using `nightly-` concatenated with the short SHA of that commit

After the nightly release, we should then delete all pre-releases except the three newest ones.

This would affect both `forgeup` and `foundry-toolchain`

### PR DESCRIPTION
**Pull request by onbjerg**
Cleans up the release workflow a bit and keeps 3 nightlies.

Nightlies are now tagged with `nightly-$SHA` as well as `nightly`. To find the latest nightly, you would look up the `nightly` tag and find the release at `nightly-$SHA` where `$SHA` is the commit that `nightly` is pointing to.

The workflow only deletes releases that have `nightly` in the tag of the release, which allows us to release other pre-release versions (betas etc.)

This also adjusts `foundryup` accordingly and there is a PR for `foundry-toolchain` too: https://github.com/onbjerg/foundry-toolchain/pull/2. **This is a breaking change for `foundryup`.**

Closes #472

### CODE REVIEW

**Comment by onbjerg** [comment]
Waiting for #488 to adjust `foundryup` so we avoid merge conflicts

**Comment by onbjerg** [comment]
I think we need a better construct for release channels. For nightly the setup in this PR works fine, but for stable it does not: the primary tag of the release is not a `stable-$SHA` tag, it's a `vx.y.z` tag. I looked at GitHub's API and there is no way for us to get all tags pointing to a specific commit unless we literally just iterate over all tags in the repository (worst case).

An alternative would be to have some file that just describes the latest release in each channel like:

```json
{
  "nightly": "nightly-$SHA",
  "stable": "v1.0.2"
}
```

The question is where to host this file.

**Comment by onbjerg** [comment]
`foundryup` now handles channels (currently only `nightly`, see above for reasoning). Running `foundryup -v nightly` will resolve the tag using GitHub's API and download the correct release after this PR is merged.

We should release a new nightly ASAP when this is merged, otherwise installations will fail

**Comment by transmissions11** [comment]
need to use the new foundry paradigm link in the foundryup dir's readme as well :)

**Comment by onbjerg** [comment]
Let me know when you think this is mergeable @gakonst and we can coordinate (need to release the GHA update around the same time)

**Comment by onbjerg** [comment]
Anything blocking this when rebased? @gakonst

**Comment by gakonst** [comment]
Can we simplify this using the new forge version which shows the commit? Cc @mattsse

**Comment by onbjerg** [comment]
The tags can't really be simplified due to how GitHub releases work, at least not without some secondary release channel to keep track of what the latest nightly is :/

**Comment by gakonst** [comment]
OK defer to you on it, let's just refactor the inline js to scripts under .github?

**Review by gakonst — APPROVED** [review]
ok let's see how this goes

**Pull request merged**
